servicetalk-http-router-jersey/src/testFixtures/java/io/servicetalk/http/router/jersey/CancellationTest.java

```diff
@@ -258,7 +259,9 @@ public void onError(final Throwable t) {
 
         cancelledLatch.await();
 
-        assertThat(errorRef.get(), is(nullValue()));
+        if (errorRef.get() != null) {
+            throw new AssertionError(errorRef.get());
```

Comment:
due to this test being racy an exception is still possible (see below). We should adjust the test so that this scenario won't fail the test.

```
java.lang.IllegalStateException: Entity input stream has already been closed.
	at io.servicetalk.http.router.jersey.DefaultJerseyStreamingHttpRouter$CloseSignalHandoffAbleContainerRequest.readEntity(DefaultJerseyStreamingHttpRouter.java:313)
	at io.servicetalk.http.router.jersey.DefaultJerseyStreamingHttpRouter$CloseSignalHandoffAbleContainerRequest.readEntity(DefaultJerseyStreamingHttpRouter.java:295)
	at org.glassfish.jersey.server.internal.inject.EntityParamValueParamProvider$EntityValueSupplier.apply(EntityParamValueParamProvider.java:73)
	at org.glassfish.jersey.server.internal.inject.EntityParamValueParamProvider$EntityValueSupplier.apply(EntityParamValueParamProvider.java:56)
	at org.glassfish.jersey.server.spi.internal.ParamValueFactoryWithSource.apply(ParamValueFactoryWithSource.java:50)
	at org.glassfish.jersey.server.spi.internal.ParameterValueHelper.getParameterValues(ParameterValueHelper.java:68)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$AbstractMethodParamInvoker.getParamValues(JavaResourceMethodDispatcherProvider.java:109)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
	at io.servicetalk.http.router.jersey.DefaultJerseyStreamingHttpRouter.handle0(DefaultJerseyStreamingHttpRouter.java:259)
	at io.servicetalk.http.router.jersey.DefaultJerseyStreamingHttpRouter.access$100(DefaultJerseyStreamingHttpRouter.java:78)
	at io.servicetalk.http.router.jersey.DefaultJerseyStreamingHttpRouter$3.handleSubscribe(DefaultJerseyStreamingHttpRouter.java:191)
	at io.servicetalk.concurrent.api.Single.handleSubscribe(Single.java:2430)
	at io.servicetalk.concurrent.api.Single.lambda$subscribeWithContext$16(Single.java:2406)
	at io.servicetalk.concurrent.api.ContextPreservingConsumer.accept(ContextPreservingConsumer.java:40)
	at io.servicetalk.concurrent.api.NoopOffloader.offloadSubscribe(NoopOffloader.java:74)
	at io.servicetalk.concurrent.api.Single.subscribeWithContext(Single.java:2405)
	at io.servicetalk.concurrent.api.Single.subscribeAndReturnContext(Single.java:2360)
	at io.servicetalk.concurrent.api.Single.subscribeInternal(Single.java:1699)
	at io.servicetalk.concurrent.api.SingleFlatMapSingle$SubscriberImpl.onSuccess(SingleFlatMapSingle.java:81)
	at io.servicetalk.concurrent.api.ContextPreservingSingleSubscriber.onSuccess(ContextPreservingSingleSubscriber.java:75)
	at io.servicetalk.concurrent.api.AbstractSubmitSingle.lambda$handleSubscribe$0(AbstractSubmitSingle.java:49)
	at io.servicetalk.concurrent.api.ContextPreservingRunnable.run(ContextPreservingRunnable.java:43)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

```diff
diff --git a/servicetalk-http-router-jersey/src/testFixtures/java/io/servicetalk/http/router/jersey/CancellationTest.java b/servicetalk-http-router-jersey/src/testFixtures/java/io/servicetalk/http/router/jersey/CancellationTest.java
index bd5f59327..79e3984d7 100644
--- a/servicetalk-http-router-jersey/src/testFixtures/java/io/servicetalk/http/router/jersey/CancellationTest.java
+++ b/servicetalk-http-router-jersey/src/testFixtures/java/io/servicetalk/http/router/jersey/CancellationTest.java
@@ -246,6 +246,9 @@ public class CancellationTest {
 
                 @Override
                 public void onError(final Throwable t) {
+                    if (t instanceof IllegalStateException) {
+                        t.printStackTrace();
+                    }
                     errorRef.compareAndSet(null, t);
                     cancelledLatch.countDown();
                 }
```